library/pk_wrap.c

```diff
@@ -1099,9 +1097,37 @@ static int eckey_check_pair(const void *pub, const void *prv,
                             int (*f_rng)(void *, unsigned char *, size_t),
                             void *p_rng)
 {
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    int ret;
+
+    mbedtls_pk_context *prv_pk = (mbedtls_pk_context *) prv;
+    mbedtls_pk_context *pub_pk = (mbedtls_pk_context *) pub;
+
+    /* When key pairs are parsed, the private part goes to the
```

Comment:
Actually, for compatibility reasons, and because past me has been stupid enough to make `mbedtls_pk_ec()` a public function (instead of declaring it in `library/pk_internal.h` or whatever), basically exposing to the whole world the `mbedtls_ecp_keypair` structure that should have been internal, we need that structure to remain present and fully populated (in builds where `ECP_C` is defined).

In the interest of keeping this PR small, I think for now we should always populate `ecp_keypair`, and keep the rest (byte array, curve family and bits) as redundant info. Then in a later PR we'll make it possible to build PK with ECC support without `ECP_C` (nor `ECP_LIGHT`) and in that config `ecp_keypair()` will be removed.

So, in the end there will be three kinds of builds:
1. Without `USE_PSA`: everything stored in `ecp_keypair` only.
2. With `USE_PSA && ECP_LIGHT`: stored in both `ecp_keypair` and new PSA format (byte array for public, slot for private) - this is redundant but I can't see another way to preserve backwards compat (other that going back in time and kindly telling past me why `mbedtls_pk_ec()` really shouldn't be public). This is where this PR should stop I think.
3. With `USE_PSA && !ECP_LIGHT`: stored only in the new PSA format (again, byte array / slot). This is not redundant, as we're not tied by backwards compatibility, since removing `ECP_C` while preserving ECC support in PK had never been possible before.

Reply:
This is something I thought about while developing this PR actually, i.e. keeping the "old" and "new" way of storing the public key present simultaneously, but then I was afraid that keeping also the "old" way around could cause me to skip some cases.
However I see your point here and the points you proposed make sense to me, so I'm going to re-elaborate this ;)